### PR TITLE
wrapper functions for fungible assets and base-class for non-fungible assets 

### PR DESCRIPTION
**Pull request by knarayan**
Changes consist of the following:

- Added non-fungible asset operations using contractId into base-class (i.e., asset_locks.go) and base-class contract (i.e., asset_locks_contract.go)
- Test cases for the changes of non-fungible asset operations (i.e., changes into asset_locks_test.go)
- Wrapper functions for the fungible/token assets (i.e., changes into assetmgmt.go)
- Refactoring of the base class code
- Minor edits to few other files

### CODE REVIEW

File: core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks_contract.go [hunk 1]
```diff
@@ -276,30 +253,46 @@ func (amc *AssetManagementContract) ClaimFungibleAsset(ctx contractapi.Transacti
...
+                ContractId: contractId,
+                Claim: claimInfoHTLC,
+            }
+            contractInfoBytes, err := proto.Marshal(contractInfo)
+            if err == nil {
+                err = ctx.GetStub().SetEvent("ClaimAssetUsingContractId", contractInfoBytes)
+            }
+        }
+        if err != nil {
+	    logWarnings("Unable to set 'ClaimAssetUsingContractId' event", err.Error())
```
Comment (VRamakrishna) [inline core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks_contract.go:287]: We can just set this event name to `ClaimAsset`. The event receiver will see the same structure regardless of whether it contains a contract id or not, and can accordingly perform suitable actions.

File: core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks_contract.go [hunk 2]
```diff
@@ -314,32 +307,51 @@ func (amc *AssetManagementContract) UnlockAsset(ctx contractapi.TransactionConte
...
+
+    // The below 'SetEvent' should be the last in a given transaction (if this function is being called by another), otherwise it will be overridden
+    retVal, err := amc.assetManagement.UnlockAssetUsingContractId(ctx.GetStub(), contractId)
+    if retVal && err == nil {
+        contractInfo := &common.AssetContractHTLC{
+            ContractId: contractId,
+        }
+        contractInfoBytes, err := proto.Marshal(contractInfo)
+        if err == nil {
+            err = ctx.GetStub().SetEvent("UnlockAssetUsingContractId", contractInfoBytes)
```
Comment (VRamakrishna) [inline core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks_contract.go:351]: We can just set this event name to `UnlockAsset`. The event receiver will see the same structure regardless of whether it contains a contract id or not, and can accordingly perform suitable actions.

File: core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks_contract.go (hunk 1 above)
Comment (knarayan) [reply core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks_contract.go]: Done @VRamakrishna .

File: core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks_contract.go (hunk 2 above)
Comment (knarayan) [reply core/network/fabric-interop-cc/interfaces/asset-mgmt/asset_locks_contract.go:351]: Done @VRamakrishna !

**Review by VRamakrishna — APPROVED** [review]

**Pull request merged**